graphql/schemabuilder/pagination.go

```diff
@@ -17,6 +17,54 @@ type Connection struct {
 	PageInfo   PageInfo
 }
 
+// paginateManually applies the pagination arguments to the edges in memory and sets hasNextPage +
+// hasPrevPage. The behavior is expected to conform to the Relay Cursor spec:
+// https://facebook.github.io/relay/graphql/connections.htm#EdgesToReturn()
+func (c *Connection) paginateManually(args PaginationArgs) error {
+	var elemsAfter, elemsBefore bool
+	c.Edges, elemsAfter, elemsBefore = applyCursorsToAllEdges(c.Edges, args.Before, args.After)
+
+	c.PageInfo.HasNextPage = args.Before != nil && elemsAfter
+	c.PageInfo.HasPrevPage = args.After != nil && elemsBefore
+
+	if (safeInt64Ptr(args.First) < 0) || safeInt64Ptr(args.Last) < 0 {
+		return graphql.NewClientError("first/last cannot be a negative integer")
+	}
+
+	if args.First != nil && args.Last != nil {
+		return graphql.NewClientError("Cannot use both first and last together")
```

Comment:
nit: lower?

Reply:
_fine_. Where does this get exposed?

Reply:
(as in: is it in context for an individual field? fragment sentence?)